docs/src/main/asciidoc/qute-reference.adoc

```diff
@@ -619,6 +619,97 @@ You can also add any number of `else` blocks:
 {/if}
 ----
 
+[[when_section]]
+==== When/Switch Section
+
+This section is similar to Java's `switch` or Kotlin's `when` constructs.
+It matches a _tested value_ against all blocks sequentially until a condition is satisfied.
+The first matching block is executed.
+All other blocks are ignored (this behavior differs to the Java `switch` where a `break` statement is necessary).
+
+.Example using the `when`/`is` name aliases
+[source]
+----
+{#when items.size}
+  {#is 1} <1>
+    There is exactly one item!
+  {#is > 10} <2>
+    There are more than 10 items!
+  {#else} <3>
+    There are 2 -10 items!
+{/when}
+----
+<1> If there is exactly one parameter it's tested for equality.
+<2> It's possible to use <<when_operators,an operator>> to specify the matching logic. Unlike in the <<if_section>> nested operators are not supported.
+<3> `else` is block is executed if no other block matches the value.
+
+.Example using the `switch`/`case` name aliases
+[source]
+----
+{#switch person.name}
+  {#case 'John'} <1>
+    Hey John!
+  {#case 'Mary'} <1>
+    Hey Mary!
+{/switch}
+----
+<1> `case` is an alias for `is`.
+
+A tested value that resolves to an enum is handled specifically.
+The parameters of an `is`/`case` block are not evaluated as expressions but compared with the result of `toString()` invocation upon the tested value.
+
+[source]
+----
+{#when machine.status}
+  {#is ON} 
+    It's running. <1>
+  {#is in OFF BROKEN}
+    It's broken or OFF. <2>
+{/when}
+----
+<1> This block is executed if `machine.status.toString().equals("ON")`.
+<2> This block is executed if  `machine.status.toString().equals("OFF")` or `machine.status.toString().equals("BROKEN")`.
+
+NOTE: The enum constants are validated if there is a parameter declaration mapped to a tested value and the declared type is an enum.
+
+The following operators are supported in `is`/`case` block conditions:
```

Comment:
I get why you have special operators, because the LHS is implicit, but we should really fix expressions so that they're not tag-dependent. For example I expect people will want to write `{#is 2 || > 20}` and `{#is (other.fu + stuff.bar)}` and I'm not sure there's a good reason to prevent that. Besides the implicit LHS.

Hell, I wonder if we shouldn't _also_ allow non-implitic operators like `{#case 2 || something == fu}`.

Reply:
Yes, we could do a lot of things but the current API is probably not the best fit. I'm all for continuous refactoring and improvements though!